openfeature/provider/registry.py

```diff
@@ -50,10 +53,31 @@ def clear_providers(self) -> None:
 
     def shutdown(self) -> None:
         for provider in {self._default_provider, *self._providers.values()}:
-            provider.shutdown()
+            self._shutdown_provider(provider)
 
     def _get_evaluation_context(self) -> EvaluationContext:
         # imported here to avoid circular imports
         from openfeature.api import get_evaluation_context
 
         return get_evaluation_context()
+
+    def _initialize_provider(self, provider: FeatureProvider) -> None:
+        try:
+            if hasattr(provider, "initialize"):
+                provider.initialize(self._get_evaluation_context())
+            self._set_provider_status(provider, ProviderStatus.READY)
+        except Exception:
```

Comment:
It should be possible for the provider to signal that the thrown exception is not recoverable (FATAL): https://github.com/open-feature/spec/blob/37cf68b0d68b6814514bcded521b9e199efcead3/specification/sections/02-providers.md?plain=1#L195

Here's something similar in JS: https://github.com/open-feature/js-sdk/blob/97a34b630eb7caa7ad744034ae56adf0f16e5005/packages/shared/src/open-feature.ts#L282-L288

Doesn't need to be done now but it seems relevant.